app/assets/javascripts/oxalis/model/sagas/skeletontracing_saga.js

```diff
@@ -121,6 +123,33 @@ function* watchFailedNodeCreations(): Saga<void> {
   }
 }
 
+function* watchTracingConsistency(): Saga<void> {
+  const state = yield* select(_state => _state);
+  const invalidTreeDetails = [];
+
+  for (const tree: Tree of _.values(enforceSkeletonTracing(state.tracing).trees)) {
+    const edgeCount = tree.edges.size();
+    const nodeCount = tree.nodes.size();
+    // For a tree, edge_count = node_count - 1 should hold true
```

Comment:
I forgot about cyclic trees when we were talking about this :/ In those, edgeCount can be equal to or even bigger than the nodeCount.
However, the `edgeCount < nodeCount - 1` condition should definitely be an indicator of broken trees, so that is correct, only the comment should be adjusted :)